### ISSUE TITLE
Implementations missings

### ISSUE THREAD

**Issue by MikaelMayer** [dafny-lang/dafny#3071]
Places where INodes was missing a children: 

* ActualBinding: `public IEnumerable<INode> Children => new List<INode> { Actual`
* Bottom-Up visitor is not visiting the by-method body of a function.
* Sub-expressions of a TypeRHS are not iterated via SubExpressions (Arguments)
* The VarDecl does not refer to the sub expressions of the resolved update statement
* If there were expressions in the LHS of a UpdateStmt or a AssignOrReturnStmt, they were not visited
* The SyntaxTreeVisitor wasn’t visiting AliasModuleDeclarations, wasn’t visiting the bound variables of a forall statement

If you find that these can cause soundness issues, please feel free to explore.

**Comment by RustanLeino** [comment]
Just looking at your list (and not the code), the one that stands out to me as a potential problem is the subexpressions of TypeRHS. Perhaps they are visited as part of assignment statements? Or this might be a bug.

**Comment by cpitclaudel** [comment]
In a few of these cases you mention resolved subexpressions: isn't the contract there that subexpressions are only to be looked at before/during resolution?

**Comment by MikaelMayer** [comment]
Yes, indeed I checked, they are visited as part of assignment statements that are attached to the TypeRHS. So it's not that much of a concern.

Basically, I found all these errors while trying to use subexpressions on a pre-resolved AST.
